server/sonar-process-monitor/src/main/java/org/sonar/application/process/EsProcessMonitor.java

```diff
@@ -0,0 +1,224 @@
+import static org.sonar.application.process.EsProcessMonitor.Status.RED;
+import static org.sonar.application.process.EsProcessMonitor.Status.YELLOW;
+
+public class EsProcessMonitor extends AbstractProcessMonitor {
+  private static final Logger LOG = LoggerFactory.getLogger(EsProcessMonitor.class);
+  private static final int WAIT_FOR_UP_DELAY_IN_MILLIS = 100;
+  private static final int WAIT_FOR_UP_TIMEOUT = 10 * 60; /* 1min */
+
+  private final AtomicBoolean nodeUp = new AtomicBoolean(false);
+  private final AtomicBoolean nodeOperational = new AtomicBoolean(false);
+  private final EsCommand esCommand;
+  private AtomicReference<TransportClient> transportClient = new AtomicReference<>(null);
+
+  public EsProcessMonitor(Process process, EsCommand esCommand) throws MalformedURLException {
+    super(process);
+    this.esCommand = esCommand;
+  }
+
+  @Override
+  public boolean isOperational() {
+    boolean operational = isOperationalImpl();
+    LOG.warn("Daniels Debugging: ES operational? "+operational);
+    return operational;
+  }
+
+  private boolean isOperationalImpl() {
+    if (nodeOperational.get()) {
+      return true;
+    }
+
+    boolean flag = false;
+    try {
+      flag = checkOperational();
+    } catch (InterruptedException e) {
+      LOG.trace("Interrupted while checking ES node is operational", e);
+      Thread.currentThread().interrupt();
+    } finally {
+      if (flag) {
+        transportClient.set(null);
+        nodeOperational.set(true);
+      }
+    }
+    return nodeOperational.get();
+  }
+
+  private boolean checkOperational() throws InterruptedException {
+    int i = 0;
+    Status status = checkStatus();
+    do {
+      if (status != Status.CONNECTION_REFUSED) {
+        nodeUp.set(true);
+      } else {
+        Thread.sleep(WAIT_FOR_UP_DELAY_IN_MILLIS);
+        i++;
+        status = checkStatus();
+      }
+    } while (!nodeUp.get() && i < WAIT_FOR_UP_TIMEOUT);
+    return status == YELLOW || status == GREEN;
+  }
+
+  static class MinimalTransportClient extends TransportClient {
+
+    MinimalTransportClient(Settings settings) {
+      super(settings, unmodifiableList(singletonList(Netty4Plugin.class)));
+    }
+
+    @Override
+    public void close() {
+      super.close();
+      if (NetworkModule.TRANSPORT_TYPE_SETTING.exists(settings) == false
+          || NetworkModule.TRANSPORT_TYPE_SETTING.get(settings).equals(Netty4Plugin.NETTY_TRANSPORT_NAME)) {
+        try {
+          GlobalEventExecutor.INSTANCE.awaitInactivity(5, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+          Thread.currentThread().interrupt();
+        }
+        try {
+          ThreadDeathWatcher.awaitInactivity(5, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+          Thread.currentThread().interrupt();
+        }
+      }
+    }
+
+  }
+
+  private Status checkStatus() {
+    Status status = checkStatusImpl();
+    LOG.warn("Daniels Debugging: ES status? "+status);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use the built-in formatting to construct this argument. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2629)